pysat/utils/time.py

```diff
@@ -269,57 +269,38 @@ def create_datetime_index(year=None, month=None, day=None, uts=None):
 
     """
 
-    # We need a timeseries index for storing satellite data in pandas, but
-    # creating a datetime object for everything is too slow.  Instead, we
-    # calculate the number of nanoseconds elapsed since first sample and
-    # create timeseries index from that.  This yields a factor of 20
-    # improvement compared to previous method, which itself was an order of
-    # magnitude faster than datetime.
-
     # Get list of unique year, and month
     if not hasattr(year, '__iter__'):
         raise ValueError('Must provide an iterable for all inputs.')
     if len(year) == 0:
         raise ValueError('Length of array must be larger than 0.')
 
-    year = year.astype(int)
+    # Establish default month
     if month is None:
-        month = np.ones(len(year), dtype=int)
-    else:
-        month = month.astype(int)
-
-    if uts is None:
-        uts = np.zeros(len(year))
-    if day is None:
-        day = np.ones(len(year))
-    day = day.astype(int)
-
-    # Track changes in seconds
-    uts_del = uts.copy().astype(np.float64)
+        # If no month, assume January.  All days will be treated as day of year.
+        month = np.ones(shape=len(year))
 
-    # Determine where there are changes in year and month that need to be
-    # accounted for
-    _, idx = np.unique((year * 100.0 + month), return_index=True)
+    # Initial day is first of given month.
+    day0 = np.ones(shape=len(year))
 
-    # Create another index array for faster algorithm below
-    idx2 = np.hstack((idx, len(year) + 1))
+    if day is None:
+        # If no day, assume first of month.
+        day = day0
+    if uts is None:
+        # If no seconds, assume start of day.
+        uts = np.zeros(shape=len(year))
 
-    # Computes UTC seconds offset for each unique set of year and month
-    for _idx, _idx2 in zip(idx[1:], idx2[2:]):
-        temp = (dt.datetime(year[_idx], month[_idx], 1)
-                - dt.datetime(year[0], month[0], 1))
-        uts_del[_idx:_idx2] += temp.total_seconds()
+    # Initialize all dates as first of month and convert to index.
+    df = pds.DataFrame({'year': year, 'month': month, 'day': day0})
```

Comment:
```suggestion
    df = pds.DataFrame({'year': year, 'month': month, 'day': day})
```

Reply:
Ah, I see the addition of days below with the index += (day - 1) thing. Nevermind.

Reply:
day could be day of year, and the routine breaks for Jan 32.

Reply:
Good catch.  Probably want to use a string to cast this as a datetime object, and then to a data frame.

Reply:
I'm using the dataframe method to avoid loops and if/then statements since the input may have month/day or day of year.  The method should be fairly resilient, but perhaps needs better comments.

Reply:
Thanks for the reminder that day could be day of year. We do want to preserve that.

Reply:
Double-checked, and it's captured in the docstring, which was unchanged.

Reply:
Shoulda been in video form since I'm young at heart 🐬